api/src/main/java/me/lokka30/treasury/api/economy/account/BankAccount.java

```diff
@@ -116,4 +120,32 @@ void hasPermission(
             @NotNull EconomySubscriber<Boolean> subscription
     );
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    default void retrieveTransactionHistory(@NotNull EconomySubscriber<Collection<? extends AccountTransaction>> subscription) {
```

Comment:
Shouldn't we just override this method and make the subscription a `EconomySubscriber<Collection<? extends BankAccountTransaction>>`? then, remove the other method?

Reply:
I will try doing this but I don't know if it will work because that's what i was trying to do at first but didn't work for some reason.

Reply:
not possible - we're staying as is

![image](https://img.mrivanplays.com/whhiivvzpr.png)

Reply:
That sucks. Is this because the subscriber is a `Collection<? extends AccountTransaction>`?

I am unsure why we are using `<? extends ..` actually, wouldn't `Collection<AccountTransaction>` work properly?

Reply:
no because it shall also accept `BankAccountTransaction`s. Generics are stuff full of compromises and pitfalls.